### PR TITLE
Replace libsmacker with libsmackerdec

### PR DESCRIPTION
**Pull request by StephenCWills**
* Stream Smacker videos from disk
  * Solves OOM errors on o3DS (3dsx) when playing the intro video
* More liberal license

### CODE REVIEW

File: Source/storm/storm_svid.cpp [hunk 1]
```diff
@@ -240,28 +237,27 @@ bool SVidPlayBegin(const char *filename, int flags)
...
+		orig_palette[i].g = paletteData[i * 3 + 1];
+		orig_palette[i].b = paletteData[i * 3 + 2];
+	}
+	memcpy(logical_palette, orig_palette, sizeof(logical_palette));
 
-		if (SDLC_SetSurfaceAndPaletteColors(SVidSurface.get(), SVidPalette.get(), colors, 0, 256) <= -1) {
-			Log("{}", SDL_GetError());
-			return false;
-		}
+	if (SDLC_SetSurfaceAndPaletteColors(SVidSurface.get(), SVidPalette.get(), colors, 0, 256) <= -1) {
```
Comment (AJenbo) [inline Source/storm/storm_svid.cpp:258]: doing this every frame may severely hurt performance on some devices

File: Source/storm/storm_svid.cpp [hunk 2]
```diff
@@ -151,36 +155,24 @@ bool SVidPlayBegin(const char *filename, int flags)
 	//0x200800 // Clear FB
 
 	SDL_RWops *videoStream = OpenAsset(filename);
-#ifdef DEVILUTIONX_SDL_RWOPS_FILE_WRAPPER_AVAILABLE
```
Comment (AJenbo) [inline Source/storm/storm_svid.cpp:154]: you can remove storm_file_wrapper.cpp as well

File: Source/storm/storm_svid.cpp (hunk 1 above)
Comment (StephenCWills) [reply Source/storm/storm_svid.cpp:258]: I was wondering about that. It's easy enough to expose a Boolean from libsmackerdec whenever the palette changes.

**Review by glebm — APPROVED** [review]
Nice!

**Comment by glebm** [comment]
I'm getting a few compiler warnings:

```
[192/200] Building CXX object 3rdParty/libsmackerdec/CMakeFiles/libsmackerdec.dir/__/__/_deps/libsmackerdec-src/src/SmackerDecoder.cpp.o
_deps/libsmackerdec-src/src/SmackerDecoder.cpp:241:21: warning: multi-character character constant [-Wmultichar]
  241 | const int kSMK2iD = 'SMK2';
      |                     ^~~~~~
_deps/libsmackerdec-src/src/SmackerDecoder.cpp:242:21: warning: multi-character character constant [-Wmultichar]
  242 | const int kSMK4iD = 'SMK4';
      |                     ^~~~~~
[199/200] Building CXX object CMakeFiles/libdevilutionx.dir/Source/storm/storm_svid.cpp.o
../Source/storm/storm_svid.cpp: In function ‘bool devilution::{anonymous}::SVidLoadNextFrame()’:
../Source/storm/storm_svid.cpp:132:14: warning: unused variable ‘frameIndex’ [-Wunused-variable]
  132 |         auto frameIndex = Smacker_GetNextFrame(SVidHandle);
      |              ^~~~~~~~~~
../Source/storm/storm_svid.cpp: In function ‘bool devilution::SVidPlayBegin(const char*, int)’:
../Source/storm/storm_svid.cpp:149:14: warning: unused variable ‘enableVideo’ [-Wunused-variable]
  149 |         bool enableVideo = (flags & 0x100000) == 0;
      |              ^~~~~~~~~~~
../Source/storm/storm_svid.cpp:165:31: warning: unused variable ‘MaxSmkChannels’ [-Wunused-variable]
  165 |         constexpr std::size_t MaxSmkChannels = 7;
      |                               ^~~~~~~~~~~~~~
../Source/storm/storm_svid.cpp:193:24: warning: unused variable ‘nFrames’ [-Wunused-variable]
  193 |         const uint32_t nFrames = Smacker_GetNumFrames(SVidHandle);
      |                        ^~~~~~~
../Source/storm/storm_svid.cpp:200:14: warning: unused variable ‘frameIndex’ [-Wunused-variable]
  200 |         auto frameIndex = Smacker_GetNextFrame(SVidHandle);
      |              ^~~~~~~~~~
```

**Review by glebm — CHANGES_REQUESTED** [review]
Please fix the warnings, looks good otherwise!

**Pull request merged**